### DISCUSSION TITLE
Feature: Plugins (for clips and chapters)

### DISCUSSION THREAD

**Discussion by subpublic** [muxinc/elements#469] [Ideas]
### Which Mux Elements/Packages would you like a new feature for?

- mux-video
- mux-audio

### Describe the feature you'd like to request

I would like being able to show clips (parts of a Mux VOD file) and display chapters in the timeline.
I could also just have a good way to add/use plugins, so that we can write one (and maybe pull request).

### Describe the use cases the feature will solve

- Show short recordings from long live stream (without have to clip the file it self).
- Guiding a viewer through long videos with chapters

### How are you using Mux Elements?

I'm not right now, 

### Describe the solution you'd like to see

For clips we're now using videojs-offset, which works fine.
For chapters I like de design of youtube and would like to be able to set chapters using JSON or JavaScript (not a vtt file)
For plugins I'd like a guide and a forkable project that includes testing (not to break anything)

## Describe alternatives or workarounds you've considered

Using VideoJS, but the mux player looks much nicer!

**Comment by subpublic** [comment]
I can media-chrome has a `<media-clip-selector>` to create a clip, but does Mux Player support that? (Can't find a way)

**Comment by dylanjha** [comment]
Mux Player does not have this directly, using `<media-clip-selector>` with Media Chrome is your best bet.

Also when using Media Chrome with Mux, you'll want to use `<mux-video>` (not Mux Player) as the media slot, so something like:

```html
<media-controller>
  <mux-video playback-id="xx" slot="media"></mux-video>
</media-controller>
```

Let me know if that makes sense

**Comment by luwes** [comment]
@subpublic you might also want to take a look at https://github.com/muxinc/media-offset which should be a valid replacement to videojs-offset. `media-offset` supports all (custom) video elements with a HTMLMediaElement API.

Have a look at https://github.com/muxinc/media-offset for the repo.
Here for some examples https://media-offset-mux.vercel.app/examples/mux-player.html

**Reply by subpublic** [reply]
Great, this is what I was waiting for. But didn't work on my iPhone, I'll leave an Issue.
